### PR TITLE
1242 erc 20 autoapproval

### PR DESCRIPTION
**Pull request by juanfri-web3**
https://github.com/xchainjs/xchainjs-lib/issues/1242

### CODE REVIEW

File: packages/xchain-aggregator/src/protocols/mayachain/mayachainProtocol.ts [hunk 1]
```diff
@@ -34,6 +34,26 @@ export class MayachainProtocol implements IProtocol {
...
     this.configuration = configuration
   }
+  /**
+   * Aprove tx for ERC-20
+   * @param {ApproveParams} approveParams params to approve tx
+   * @returns {TxSubmitted} Transaction hash and URL of the swap
+   */
+  async approveRouterToSpend(params: ApproveParams): Promise<TxSubmitted> {
+    const { asset, amount } = params
+    return await this.mayachainAmm.approveRouterToSpend({ asset, amount })
```
Comment (0xp3gasus) [inline packages/xchain-aggregator/src/protocols/mayachain/mayachainProtocol.ts:44]: We need to be sure approve transaction has been confirmed.

File: .changeset/hungry-eggs-move.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (0xp3gasus) [inline .changeset/hungry-eggs-move.md:1]: Split Changeset file into this ones to be more precise in the changelog files:
- xchain-util -> New isTokenAssetFunction
- xchain-evm and xchain-wallet -> Add function `awaitTxConfirmed` which waits until the approval is confirmed
- xchain-thorchain-amm, xchain-mayachain-amm -> ERC20 check for isRouterApprovedToSpend
- xchain-aggregator -> Support auto-approval for ERC-20 tokens

Comment (juanfril) [reply .changeset/hungry-eggs-move.md:1]: Perfect, thank you!

**Review by 0xp3gasus — APPROVED** [review]
Great work!

**Review by 0xp3gasus — APPROVED** [review]

**Pull request merged**